src/core/playground/signer.ts

```diff
@@ -0,0 +1,66 @@
+import { HexInput } from "../../types";
+import { AccountAddress } from "../accountAddress";
+import { AccountAuthenticatorSingleKey } from "./accountAuthenticator";
+import { Ed25519PrivateKey, type Ed25519PublicKey, type Ed25519Signature } from "./ed25519";
+import type { BaseSigner, PrivateKey, PublicKey } from "./interfaces";
+import { SignatureScheme } from "./scheme";
+import { Secp256k1PrivateKey, type Secp256k1PublicKey, type Secp256k1Signature } from "./secp256k1";
+import { AllowedSignatures, WrappedPublicKey, WrappedSignature } from "./wrapped";
+
+export interface SignerConstructorArgs<
+  TSignature extends AllowedSignatures,
+  TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+> {
+  privateKey: PrivateKey<TSignature, TPublicKey>;
+  address?: AccountAddress | HexInput;
+}
+
+export interface GenerateSignerArgs<TSignatureScheme extends SignatureScheme = SignatureScheme> {
+  scheme: TSignatureScheme;
+}
+
+export class Signer<
+  TSignature extends AllowedSignatures = AllowedSignatures,
+  TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+> implements BaseSigner<AccountAuthenticatorSingleKey<TSignature, TPublicKey>>
```

Comment:
Signer could be good over an Account.  Especially if you could sign as an object in the future...

Reply:
what do you mean by "signing as an object"?

Reply:
Today the only things that can sign are accounts, but imagine other things signing transactions

Reply:
gotcha, interesting